README.md
# `is-dom-node-list`

<a href="https://travis-ci.org/jlmakes/is-dom-node-list"> <img src="https://img.shields.io/travis/jlmakes/is-dom-node-list.svg" alt="Build status"> </a>
<a href="https://coveralls.io/github/jlmakes/is-dom-node-list"> <img src="https://img.shields.io/coveralls/jlmakes/is-dom-node-list.svg" alt="Coverage"> </a>
<a href="https://www.npmjs.com/package/is-dom-node-list"> <img src="https://img.shields.io/npm/v/is-dom-node-list.svg" alt="Version"> </a>
<img src="https://img.shields.io/badge/min+gzip-0.3KB-blue.svg" alt="0.3KB min+gzip">
<a href="https://github.com/jlmakes/is-dom-node-list/blob/master/LICENSE"> <img src="https://img.shields.io/badge/license-MIT-1283c3.svg" alt="MIT License"> </a>

## Installation

### Browser

A simple and fast way to get started is to include this script on your page:

```html
<script src="https://unpkg.com/is-dom-node-list"></script>
```

This will create the global variable `isDomNodeList`.

### Module

```bash
npm install is-dom-node-list
```

#### CommonJS

```js
const isDomNodeList = require('is-dom-node-list')
```

#### ES2015

```js
import isDomNodeList from 'is-dom-node-list'
```

## Usage

```js
const nodeList = document.querySelectorAll('.cookies')
isDomNodeList(nodeList)
// => true

const nodeArray = Array.prototype.slice.call(nodeList)
isDomNodeList(nodeArray)
// => false
```


CHANGELOG.md
# Change Log

## Unreleased

* Revised readme.
* Upgrade `is-dom-node` dependency.

## [1.2.0] - 2018-01-11

### Changed

* `NodeList` detection algorithm reverted.
* Package once again relies on `is-dom-node`.
* The ES distribution does not bundle `is-dom-node`.
* Revised readme.

## [1.1.0] - 2018-01-11

### Changed

* `NodeList` detection algorithm has been updated.
* Package no longer relies on `is-dom-node`.

## [1.0.2] - 2018-01-10

### Changed

* Updated repository file structure.

## [1.0.1] - 2018-01-06

### Fixed

* Updated test suite formatting.

## [1.0.0] - 2018-01-06

Hello world!

[1.2.0]: https://github.com/jlmakes/is-dom-node-list/compare/1.1.0...1.2.0
[1.1.0]: https://github.com/jlmakes/is-dom-node-list/compare/1.0.2...1.1.0
[1.0.2]: https://github.com/jlmakes/is-dom-node-list/compare/1.0.1...1.0.2
[1.0.1]: https://github.com/jlmakes/is-dom-node-list/compare/1.0.0...1.0.1
[1.0.0]: https://github.com/jlmakes/is-dom-node-list/tree/1.0.0


index.test.js
import isDomNodeList from './index'

describe('/utils/', () => {
	describe('isNodeList(target)', () => {
		it('should return true when passed a NodeList', () => {
			const result = isDomNodeList(document.querySelectorAll('script'))
			expect(result).to.be.true
		})

		it('should return false when passed an Array<Node>', () => {
			const elements = document.querySelectorAll('body')
			const result = isDomNodeList(Array.prototype.slice.call(elements))
			expect(result).to.be.false
		})

		it('should return false when passed an invalid target', () => {
			expect(isDomNodeList(null)).to.be.false
			expect(isDomNodeList()).to.be.false
			expect(isDomNodeList({})).to.be.false
			expect(isDomNodeList('!')).to.be.false
		})
	})
})


test/index.test.js
import isDomNodeList from '../src/index'

describe('isDomNodeList(target)', () => {
	it('should return true when passed a NodeList', () => {
		const result = isDomNodeList(document.querySelectorAll('script'))
		expect(result).to.be.true
	})

	it('should return false when passed an Array<Node>', () => {
		const elements = document.querySelectorAll('body')
		const result = isDomNodeList(Array.prototype.slice.call(elements))
		expect(result).to.be.false
	})

	it('should return false when passed an invalid target', () => {
		expect(isDomNodeList(null)).to.be.false
		expect(isDomNodeList()).to.be.false
		expect(isDomNodeList({})).to.be.false
		expect(isDomNodeList('!')).to.be.false
	})
})


test/karma.conf.js
const rollupPlugins = [
	require('rollup-plugin-buble')(),
	require('rollup-plugin-istanbul')({
		exclude: ['index.test.js', '**/node_modules/**'],
		instrumenterConfig: {
			embedSource: true // helps with inaccurate line numbers
		}
	}),
	require('rollup-plugin-json')(),
	require('rollup-plugin-node-resolve')()
]

module.exports = karma => {
	karma.set({
		frameworks: ['mocha', 'sinon-chai'],

		preprocessors: {
			'index.test.js': ['rollup']
		},

		files: [{ pattern: 'index.test.js', watched: false }],

		rollupPreprocessor: {
			plugins: rollupPlugins,
			format: 'iife',
			name: 'isDomNodeList',
			sourcemap: 'inline'
		},

		colors: true,
		concurrency: 10,
		logLevel: karma.LOG_ERROR,
		singleRun: true,

		browsers: ['ChromeHeadless'],
		browserDisconnectTolerance: 1,
		browserDisconnectTimeout: 60 * 1000,
		browserNoActivityTimeout: 60 * 1000,
		captureTimeout: 4 * 60 * 1000
	})

	if (process.env.TRAVIS) {
		karma.set({
			autoWatch: false,
			coverageReporter: {
				type: 'lcovonly',
				dir: 'coverage/'
			},
			reporters: ['mocha', 'coverage', 'coveralls']
		})
	} else {
		karma.set({
			coverageReporter: {
				type: 'lcov',
				dir: '../.ignore/coverage/'
			},
			reporters: ['dots', 'coverage']
		})
	}
}


rollup.conf.banner.js
const { version } = require('./package.json')

export const condensed = `/*! @license is-dom-node-list v${version} (MIT) Copyright 2018 Fisssion LLC. */`

export const standard = `/*! @license is-dom-node-list v${version}

	Copyright 2018 Fisssion LLC.

	Permission is hereby granted, free of charge, to any person obtaining a copy
	of this software and associated documentation files (the "Software"), to deal
	in the Software without restriction, including without limitation the rights
	to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
	copies of the Software, and to permit persons to whom the Software is
	furnished to do so, subject to the following conditions:

	The above copyright notice and this permission notice shall be included in all
	copies or substantial portions of the Software.

	THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
	IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
	FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
	AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
	LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
	OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
	SOFTWARE.

*/`


karma.conf.js
const rollupPlugins = [
	require('rollup-plugin-json')(),
	require('rollup-plugin-node-resolve')({ jsnext: true, main: true }),
	require('rollup-plugin-buble')(),
	require('rollup-plugin-istanbul')({
		exclude: ['index.test.js', '**/node_modules/**'],
		instrumenterConfig: {
			embedSource: true // helps with inaccurate line numbers
		}
	})
]

module.exports = karma => {
	karma.set({
		frameworks: ['mocha', 'sinon-chai'],

		preprocessors: {
			'index.test.js': ['rollup']
		},

		files: [{ pattern: 'index.test.js', watched: false }],

		rollupPreprocessor: {
			plugins: rollupPlugins,
			format: 'iife',
			name: 'isDomNodeList',
			sourcemap: 'inline'
		},

		colors: true,
		concurrency: 10,
		logLevel: karma.LOG_ERROR,
		singleRun: true,

		browsers: ['ChromeHeadless'],
		browserDisconnectTolerance: 1,
		browserDisconnectTimeout: 60 * 1000,
		browserNoActivityTimeout: 60 * 1000,
		captureTimeout: 4 * 60 * 1000
	})

	if (process.env.TRAVIS) {
		karma.set({
			autoWatch: false,
			coverageReporter: {
				type: 'lcovonly',
				dir: 'coverage/'
			},
			reporters: ['mocha', 'coverage', 'coveralls']
		})
	} else {
		karma.set({
			coverageReporter: {
				type: 'lcov',
				dir: '../.ignore/coverage/'
			},
			reporters: ['dots', 'coverage']
		})
	}
}
